paasta_tools/mesos_maintenance.py

```diff
@@ -266,6 +288,29 @@ def get_machine_ids(hostnames):
     return machine_ids
 
 
+def build_reservation_payload(slave_id, resource, amount):
+    """Creates the JSON payload needed to dynamically (un)reserve resources in mesos.
+    :param slave_id: the id of the mesos slave
+    :param resource: the mesos resource type we want to reserve (i.e. cpus, mem, gpus, etc)
+    :param amount: integer representing the amount of the above resource to reserve
+    :returns: a dictionary that can be sent to Mesos to (un)reserve resources
+    """
+    payload = [
+        {
+            "name": resource,
+            "type": "SCALAR",
+            "scalar": {
+                "value": amount,
+            },
+            "role": "maintenance",
+            "reservation": {
+                "principal": str(get_principal()),
```

Comment:
This was me fighting against `MagicMocks`. However, I made a number of changes to the itest since then, so let me double check if it still passes without this cast.